crates/rooch/src/commands/account/commands/balance.rs

```diff
@@ -0,0 +1,139 @@
+use move_core_types::account_address::AccountAddress;
+use moveos_types::access_path::AccessPath;
+use moveos_types::module_binding::{ModuleBinding, MoveFunctionCaller};
+use moveos_types::move_option::MoveOption;
+use moveos_types::object::ObjectID;
+use moveos_types::state::MoveStructState;
+use moveos_types::state_resolver::resource_tag_to_key;
+use moveos_types::transaction::FunctionCall;
+use moveos_types::tx_context::TxContext;
+use rooch_rpc_api::api::MAX_RESULT_LIMIT_USIZE;
+use rooch_rpc_api::jsonrpc_types::account_view::AccountInfoView;
+use rooch_rpc_api::jsonrpc_types::{AccountAddressView, AnnotatedCoinStoreView, StructTagView};
+use rooch_types::account::BalanceInfo;
+use rooch_types::error::{RoochError, RoochResult};
+use rooch_types::framework::coin::CoinModule;
+
+/// Show a account info, only the accounts managed by the current node are supported
+#[derive(Debug, Parser)]
+pub struct BalanceCommand {
+    #[clap(short = 'a', long = "address")]
+    /// The account's address to show balance, if absent, show the default active account.
+    address: Option<AccountAddressView>,
+
+    /// Struct name as `<ADDRESS>::<MODULE_ID>::<STRUCT_NAME><TypeParam1?, TypeParam2?>`
+    /// Example: `0x123::counter::Counter`, `0x123::counter::Box<0x123::counter::Counter>`
+    #[clap(long = "coin_type")]
+    /// The block number or block hash for get state, if absent, use latest block state_root.
+    coin_type: Option<StructTagView>,
+
+    #[clap(flatten)]
+    pub context_options: WalletContextOptions,
+}
+
+#[async_trait]
+impl CommandAction<()> for BalanceCommand {
+    async fn execute(self) -> RoochResult<()> {
+        let context = self.context_options.build().await?;
+
+        let addr_view: Option<AccountAddress> = if let Some(address) = self.address {
+            Some(address.into())
+        } else {
+            context.config.active_address.map(|address| address.into())
+        };
+
+        // Obtain account address
+        let addr = addr_view.expect("Account not found error");
+
+        let client = context.get_client().await?;
+
+        let ctx = TxContext::new_readonly_ctx(addr);
+        let call = FunctionCall::new(
+            CoinModule::function_id(CoinModule::COIN_STORE_HANDLE_FUNCTION_NAME),
+            vec![],
+            vec![addr.to_vec()],
+        );
+        let coin_store_handle_opt: Option<ObjectID> = client
+            .call_function(&ctx, call)?
+            .into_result()
+            .map(|values| {
+                let value = values.get(0).expect("Expected return value");
+                let result = MoveOption::<ObjectID>::from_bytes(&value.value)
+                    .expect("Expected Option<ObjectID>");
+                result.into()
+            })?;
+        let coin_store_handle = coin_store_handle_opt
+            // .expect(format!("Failed to get coin store handle via {}", addr).as_str());
+            .unwrap_or_else(|| panic!("Failed to get coin store handle via {}", addr));
```

Comment:
Directly call the `coin_module::coin_store_handle`?

Reply:
I tried it before, but there are some problems.

After binding `coin_mudule`, await occurs here, with `coin_mudule` maybe used later.

the trait `Sync` is not implemented for `dyn MoveFunctionCaller`, future is not `Send` as this value is used across an await.

Reply:
The client has already implemented the `MoveFunctionCaller`.

https://github.com/rooch-network/rooch/blob/a63525896448ae9da62600c0f1fe86926f122883/crates/rooch/src/commands/session_key/commands/create.rs#L71-L79

Reply:
The difference is whether `MoveFunctionCaller` is called by async/await.

You can open these two lines of comments, cargo build to see the specific error.

https://github.com/rooch-network/rooch/blob/370bb0ed02932fdf0ad0f9ef17b66bd31e83b265/crates/rooch/src/commands/account/commands/balance.rs#L73-L74